Bottom: b9ac85522bb31e421e0097945672d782165876ff
Top:    79b5a22dab77d7d25ac8c6eb73cf55edb2593270
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 16:31:26 +0000

Adjust coordinate system GOODNES KNOWS WHETHER THE GEOMETRY IS CORRECT... IDA-STEP SEEMS TO COPE THOUGH


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index 9a3a626..63ea6bf 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -24,6 +24,7 @@
 #define PERFECT_ROUND_CONTOURS
 
 #define REVERSED_PCB_CONTOURS 1 /* PCB Contours are reversed from the expected CCW for outer ordering - once the Y-coordinate flip is taken into account */
+#undef REVERSED_PCB_CONTOURS
 
 #ifdef REVERSED_PCB_CONTOURS
 #define COORD_TO_STEP_X(pcb, x) (COORD_TO_MM(                   (x)))
@@ -385,7 +386,7 @@ object3d_from_contours (const POLYAREA *contours,
 #ifdef REVERSED_PCB_CONTOURS
       face3d_set_normal (faces[npoints], 0., 0., -1.);
 #else
-      face3d_set_normal (faces[npoints], 0., 0., 1.);
+      face3d_set_normal (faces[npoints], 0., 0., 1.); /* PCB bottom is at positive Z in this scheme */
 #endif
       face3d_set_appearance (faces[npoints], top_bot_appearance);
       object3d_add_face (object, faces[npoints]);
@@ -394,7 +395,7 @@ object3d_from_contours (const POLYAREA *contours,
 #ifdef REVERSED_PCB_CONTOURS
       face3d_set_normal (faces[npoints + 1], 0., 0., 1.);
 #else
-      face3d_set_normal (faces[npoints + 1], 0., 0., -1.);
+      face3d_set_normal (faces[npoints + 1], 0., 0., -1.); /* PCB top is at negative Z in this scheme */
 #endif
       face3d_set_appearance (faces[npoints + 1], top_bot_appearance);
       object3d_add_face (object, faces[npoints + 1]);
diff --git a/src/hid/step/object3d_step.c b/src/hid/step/object3d_step.c
index 829221b..d47a8b4 100644
--- a/src/hid/step/object3d_step.c
+++ b/src/hid/step/object3d_step.c
@@ -26,6 +26,7 @@
 
 
 #define REVERSED_PCB_CONTOURS 1 /* PCB Contours are reversed from the expected CCW for outer ordering - once the Y-coordinate flip is taken into account */
+#undef REVERSED_PCB_CONTOURS
 
 #define EPSILON 1e-5 /* XXX: Unknown  what this needs to be */
